web-marketplace/src/hooks/projects/useProjectsWithOrders.ts

```diff
@@ -147,66 +148,101 @@ export function useProjectsWithOrders({
     projects = projectsData?.projects;
   }
 
-  const selectedProjects =
-    selectProjects({
-      projects,
-      sellOrders,
-      metadata,
-      random,
-      skippedProjectId,
-      skippedClassId: !classId && !projectId ? SKIPPED_CLASS_ID : undefined,
-    }) ?? [];
-
+  const skippedClassId = !classId && !projectId ? SKIPPED_CLASS_ID : undefined;
+  const selectedProjects = useMemo(
```

Comment:
using `useMemo` to avoid recomputing operations that can be expensive (filter, sort, etc.)